pkg/service/restore/schema_worker.go

```diff
@@ -273,3 +323,93 @@ func (w *schemaWorker) getFileNamesMapping(sstables []string, sstableUUIDFormat
 	}
 	return sstable.RenameToIDs(sstables, &w.generationCnt)
 }
+
+func getDescribedSchema(ctx context.Context, client *scyllaclient.Client, snapshotTag string, locHost map[Location][]string) (*query.DescribedSchema, error) {
+	baseDir := path.Join("backup", string(SchemaDirKind))
+	// It's enough to get a single schema file, but it's important to validate
+	// that each location contains exactly one or none of them.
+	var (
+		host       string
+		schemaPath *string
+		err        error
+		foundCnt   int
+	)
+	for l, hosts := range locHost {
+		host = hosts[0]
+		schemaPath, err = getRemoteSchemaFilePath(ctx, client, snapshotTag, host, l.RemotePath(baseDir))
+		if err != nil {
+			return nil, errors.Wrapf(err, "get schema file from %s", l.RemotePath(baseDir))
+		}
+		if schemaPath != nil {
+			foundCnt++
+		}
+	}
+
+	if foundCnt == 0 {
+		return nil, nil // nolint: nilnil
+	} else if foundCnt < len(locHost) {
+		return nil, errors.New("only a subset of provided locations has schema files")
+	}
+
+	r, err := client.RcloneOpen(ctx, host, *schemaPath)
```

Comment:
I think it's caller responsibility to close the Reader here.
```
// RcloneOpen streams remote file content. The stream is an HTTP body.
// Callers must close the body after use.
```
Please defer the r.Close and log error if appears.
Closing twice should not cause errors.

Reply:
Nice catch!